### PR TITLE
feat(logs): add grantRead function to LogGroup

### PR DESCRIPTION
**Pull request by watany-dev**
fixes https://github.com/aws/aws-cdk/issues/21668

adding method `logGroup.grantRead`

We refer to the following suspended PRs
https://github.com/aws/aws-cdk/pull/22132

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

* [ ] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by HBobertz — COMMENTED** [review]
Looks good to me, and mirrors the structure of grantWrite. Going to get someone else to check this out as well before I approve

**Comment by HBobertz** [comment]
@TheRealAmazonKendra

**Review by TheRealAmazonKendra — DISMISSED** [review]

**Review by HBobertz — APPROVED** [review]

File: packages/@aws-cdk/aws-logs/lib/log-group.ts [hunk 1]
```diff
@@ -169,6 +174,13 @@ abstract class LogGroupBase extends Resource implements ILogGroup {
     return this.grant(grantee, 'logs:CreateLogStream', 'logs:PutLogEvents');
   }
 
+  /**
+   * Give permissions to read and filter events from this log group
+   */
+  public grantRead(grantee: iam.IGrantable) {
+    return this.grant(grantee, 'logs:FilterLogEvents', 'logs:GetLogEvents', 'logs:GetLogRecord');
```
Comment (mrgrain) [inline packages/@aws-cdk/aws-logs/lib/log-group.ts:181]: [According to the docs](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_GetLogRecord.html), GetLogRecord is only used with a response from `GetQueryResults`. Any reason why you've included that?
```suggestion
    return this.grant(grantee, 'logs:FilterLogEvents', 'logs:GetLogEvents');
```

Comment (mrgrain) [reply packages/@aws-cdk/aws-logs/lib/log-group.ts:181]: We could probably throw a `DescribeLogGroups` and `DescribeLogStreams` in here
Also `GetLogGroupFields`

**Review by mrgrain — CHANGES_REQUESTED** [review]

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**